core/src/main/java/io/questdb/griffin/engine/QueryProgress.java

```diff
@@ -120,23 +132,30 @@ public static void logEnd(
                 log.I$();
             }
         }
+        if (queryTrace != null && engine.getConfiguration().isQueryTracingEnabled()) {
+            queryTrace.isJit = isJit;
+            queryTrace.timestamp = config.getMicrosecondClock().getTicks();
+            queryTrace.queryText = sqlText;
```

Comment:
this is not always an immutable object, sometimes it can be other times it is not. We need to find a way to copy sql text into the queue slot

Reply:
in fact the only time query trace is not null is when it is provided from QueryProgress.close(). E.g. trace belongs to the QueryProgress instance. This instance already have an immutable query text. Assigning sqlText to query trace here is confusing